docs/source/howto/data.rst

```diff
@@ -248,11 +247,247 @@ The node class has various methods to set these attributes, such as :py:`~aiida.
 
 .. _how-to:data:find:
 
-Finding data
-============
+Finding and querying for data
+=============================
+
+Once you have successfully completed a series of workflows for your project, or have imported a dataset you are interested in, you want to quickly find the data that is relevant for your analysis.
+The data in an AiiDA database is stored as a graph of connected entities, which can be easily *queried* with the ``QueryBuilder`` class.
```

Comment:
Make sure to actually link the class to the API reference (here and everywhere else).